dotnet/src/Connectors/Connectors.Amazon/Bedrock/Models/Meta/LlamaChatRequest.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Amazon.Runtime.Documents;
+using Connectors.Amazon.Core.Requests;
+
+namespace Connectors.Amazon.Models.Meta;
+
+/// <summary>
+/// Meta Llama chat completion request body.
+/// </summary>
+public class LlamaChatRequest : IChatCompletionRequest
+{
+    /// <summary>
+    /// (Required) The prompt that you want to pass to the model.
+    /// </summary>
+    public string? Prompt { get; set; }
+    /// <summary>
+    /// Use a lower value to decrease randomness in the response.
+    /// </summary>
+    public double Temperature { get; set; }
+    /// <summary>
+    /// Use a lower value to ignore less probable options. Set to 0 or 1.0 to disable.
+    /// </summary>
+    public double TopP { get; set; }
+    /// <summary>
+    /// Specify the maximum number of tokens to use in the generated response. The model truncates the response once the generated text exceeds max_gen_len.
+    /// </summary>
+    public int MaxGenLen { get; set; }
+    private List<Message>? _messages;
+
+    /// <summary>
+    /// List of messages.
+    /// </summary>
+    public List<Message> Messages
+    {
+        get
+        {
+            if (this._messages == null)
+            {
+                this._messages = new List<Message>();
+                if (!string.IsNullOrEmpty(this.Prompt))
+                {
+                    var role = ConversationRole.User;
+                    string systemPrompt = null;
+                    if (this.Prompt.StartsWith("<s>[INST]"))
+                    {
+                        var parts = this.Prompt.Split("[/INST]", StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I've copied this code to see how it would work to .net 4.8 project and it did not compile due missing `Split` overload, that would take string as first parameter. 
And the more I look at what this code is trying to do, the less I like it, looks like lots if string allocations and still hard to understand what it is trying to achieve.

Reply:
This string breakdown was originally to help parse the prompt template that Meta Llama 2 requires users to have in chat (Llama 2 is a legacy model version that will be unavailable starting Aug 12, 2024 anyway). Llama 3 just has newlines and I'm now grabbing directly from chatHistory there in the converseRequest so this is no longer needed. Thanks for bringing this to my attention!